### PR TITLE
[feat][cli] Add command line option for configuring the memory limit 

### LINKED ISSUE #15912: [Cli tools] Add command line option for configuring the memory limit
**Is your enhancement request related to a problem? Please describe.**
Cli tools such as `pulsar-perf` don't contain a way to configure the Pulsar client memory limit. 

**Describe the solution you'd like**
Cli tools should have a way to configure the Pulsar client memory limit.

**Additional context**
See https://github.com/apache/pulsar/pull/15748#issuecomment-1145013379

### PR DESCRIPTION
**Pull request by shy-share**
Fixes #15912 

Master Issue: #xyz

PIP: #xyz 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerfClientUtils.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * distributed with this work for additional information
  * regarding copyright ownership.  The ASF licenses this file
  * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
+ * "License"); you may not use this file except in cfompliance
```
Comment (lhotari) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerfClientUtils.java:7]: typo

File: pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java [hunk 2]
```diff
@@ -76,6 +76,9 @@ public static class RootParams {
 
         @Parameter(names = { "--tlsTrustCertsFilePath" }, description = "File path to client trust certificates")
         String tlsTrustCertsFilePath;
+
+        @Parameter(names = { "-m", "--memory", }, description = "Configure the Pulsar client memory limit")
+        Long memory=0L;
```
Comment (lhotari) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:81]: It's pretty unconvenient to specify megabytes as bytes on the command line. Do we already have a parser that can accept typical units like M = megabyte etc. .

Comment (shy-share) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:81]: Do you think it should be changed to this ？

@Parameter(names = { "-M", "--Memory", },

Comment (tisonkun) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:81]: @shy-share IIUC it means you can specify `-m 41M` means configuring memory as 41MB.

Comment (shy-share) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:81]: @tisonkun I know what you mean but I don't know how to write this

Comment (tisonkun) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:81]: @lhotari @nicoloboschi Does our CLI tools already provide such functionality? I can read `broker.conf` options like:

```ini
# For Amazon S3 ledger offload, Max block size in bytes. (64MB by default, 5MB minimum)
s3ManagedLedgerOffloadMaxBlockSizeInBytes=67108864

# For Amazon S3 ledger offload, Read buffer size in bytes (1MB by default)
s3ManagedLedgerOffloadReadBufferSizeInBytes=1048576
```

**Comment by tisonkun** [comment]
I notice that we do have some utils to implement this future. See https://github.com/apache/pulsar/pull/20646#pullrequestreview-1497799011.

Closing as stale...

But anyone can take over this work as it's clear to follow now.

**Pull request closed without merging**